### PR TITLE
#33788: Make things safer when tor_inet_ntop() and tor_inet_ntoa() fail (2)

### PR DESCRIPTION
**Pull request by rl1987**
https://trac.torproject.org/projects/tor/ticket/33788

### CODE REVIEW

File: src/feature/relay/dns.c [hunk 1]
```diff
@@ -1872,16 +1878,22 @@ evdns_wildcard_check_callback(int result, char type, int count, int ttl,
       for (i = 0; i < count; ++i) {
         char answer_buf[INET_NTOA_BUF_LEN+1];
         struct in_addr in;
+        int ntoa_res;
         in.s_addr = addrs[i];
-        tor_inet_ntoa(&in, answer_buf, sizeof(answer_buf));
-        wildcard_increment_answer(answer_buf);
+        ntoa_res = tor_inet_ntoa(&in, answer_buf, sizeof(answer_buf));
+        tor_assert_nonfatal(ntoa_res > 0);
```
Comment (nmathewson) [inline src/feature/relay/dns.c:1884]: This is incorrect; look at the documentation for tor_inet_ntoa().  It says "returns -1 on failure" and "returns non-negative integer on success."  So the test should be `>= 0`.

File: src/lib/net/address.c [hunk 2]
```diff
@@ -1196,14 +1196,24 @@ fmt_addrport(const tor_addr_t *addr, uint16_t port)
...
 {
   static char buf[INET_NTOA_BUF_LEN];
   struct in_addr in;
+  int success;
+
   in.s_addr = htonl(addr);
-  tor_inet_ntoa(&in, buf, sizeof(buf));
+
+  success = tor_inet_ntoa(&in, buf, sizeof(buf));
+  tor_assertf_nonfatal(success > 0,
```
Comment (nmathewson) [inline src/lib/net/address.c:1210]: Should be >= 0.

File: src/lib/net/address.c [hunk 3]
```diff
@@ -1196,14 +1196,24 @@ fmt_addrport(const tor_addr_t *addr, uint16_t port)
...
+      "Failed to convert IP %04X to string", addr);
+
+  if (success <= 0) {
```
Comment (nmathewson) [inline src/lib/net/address.c:1213]: Should be < 0.  And may we should be using IF_BUG_ONCE for this?

File: src/feature/relay/dns.c (hunk 1 above)
Comment (rl1987) [reply src/feature/relay/dns.c]: All three check for `tor_inet_ntoa()` return value fixed.

**Pull request merged**